src/admin/shared/hoc/with-admin-core-config.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+	AdminConfig,
+	AdminConfigManager,
+	AvoOrHetArchief,
+	CommonUser,
+	ContentBlockType,
+	LinkInfo,
+	ToastInfo,
+} from '@meemoo/admin-core-ui';
+import { Icon, IconName, IconProps, Spinner } from '@viaa/avo2-components';
+import { Avo } from '@viaa/avo2-types';
+import { compact, noop } from 'lodash-es';
+import React, { ComponentType, FunctionComponent, useCallback, useEffect, useState } from 'react';
+import { withRouter } from 'react-router';
+import { Link, useHistory, useParams } from 'react-router-dom';
+import { compose } from 'redux';
+
+import { toAbsoluteUrl } from '../../../authentication/helpers/redirects';
+import { APP_PATH, RouteId } from '../../../constants';
+import { getEnv } from '../../../shared/helpers';
+import { tHtml } from '../../../shared/helpers/translate';
+import withUser from '../../../shared/hocs/withUser';
+import useTranslation from '../../../shared/hooks/use-translation/use-translation';
+import { AssetsService } from '../../../shared/services/assets-service/assets.service';
+import { SmartschoolAnalyticsService } from '../../../shared/services/smartschool-analytics-service';
+import { ToastService } from '../../../shared/services/toast-service';
+import { navigationService } from '../../navigation/services/navigation-service';
+import { PermissionsService } from '../services/permissions';
+import { UserGroupsService } from '../services/user-groups';
+
+const InternalLink = (linkInfo: LinkInfo) => {
+	return <Link {...linkInfo} />;
+};
+
+export const withAdminCoreConfig = (WrappedComponent: ComponentType): ComponentType => {
+	const Component = (props: Record<string, unknown>) => {
+		const [adminCoreConfig, setAdminCoreConfig] = useState<AdminConfig | null>(null);
+		const user = props.user as Avo.User.User | undefined;
+		const { tText } = useTranslation();
+
+		const initConfigValue = useCallback(() => {
+			const commonUser: CommonUser = {
+				uid: user?.uid,
+				profileId: user?.profile?.id as string,
+				userId: user?.uid,
+				// idp: user?.idp,
+				email: user?.mail,
+				acceptedTosAt: '1970-01-01', // TODO see where this is stored in avo
+				userGroup: {
+					name: String(user?.profile?.userGroupIds[0]), // TODO figure out a label in a synchronous way
+					id: user?.profile?.userGroupIds[0],
+				},
+				firstName: user?.first_name || undefined,
+				lastName: user?.last_name || undefined,
+				fullName: user?.full_name || undefined,
+				last_access_at: user?.last_access_at || undefined, // TODO enable once last_access_at field is added to the database
+				// eslint-disable-next-line @typescript-eslint/no-explicit-any
+				permissions: user?.profile?.permissions as any[],
+			};
+
+			const config: AdminConfig = {
```

Comment:
In general I'd move the static parts of this config into a dedicated ts file to improve formatting and legibility.